### PR TITLE
[gh-1117] add math modules.

### LINKED ISSUE #1117: [MoveosStd] Math modules
Introduce some math modules for calculating, such as `pow`,`sqrt`,`mul_div`.

### Reference

* https://github.com/starcoinorg/starcoin-framework/blob/345a3900a0064dc57a9560235bc72c12f03448b1/sources/Math.move
* https://github.com/aptos-labs/aptos-core/blob/15280ac4cbb430e81070d2df0732064631b2e23f/aptos-move/framework/aptos-stdlib/sources/math128.move
* https://github.com/MystenLabs/sui/blob/da72e73a9f4a1b42df863b76af7745a4eeb1d412/crates/sui-framework/packages/sui-framework/sources/math.move

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

Add move math modules `u8`, `u16`, `u32`, `u64`, `u128`, `u256`.

- Closes #1117

### CODE REVIEW

File: frameworks/move-stdlib/sources/macros.move [hunk 1]
```diff
@@ -0,0 +1,122 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module holds shared implementation of macros used in `std`
+module std::macros {
+    public macro fun num_max($x: _, $y: _): _ {
```
Comment (baichuan3) [inline frameworks/move-stdlib/sources/macros.move:9]: Do macros support arguments of arbitrary types?

Comment (feliciss) [reply frameworks/move-stdlib/sources/macros.move:9]: See the above doc.

> Type inference holes _ on type annotations
> 
> With type directed programming, often you need to annotate a variable declaration or provide type arguments. But, sometimes you really only need to annotate on specific type, but the other types can be inferred. _ will be added to allow that type to still be inferred, even when other parts of the type are annotated. For example
> 
> dynamic_field::borrow_mut<address, Coin<SUI>>(&mut id, owner)
> 
> could be rewritten as
> 
> dynamic_field::borrow_mut<_, Coin<SUI>>(&mut id, owner)
> 
> where the _ would be inferred as address

**Comment by jolestar** [comment]
Rooch move do not support macros yet.

**Comment by feliciss** [comment]
> Rooch move do not support macros yet.

Is there a way to support the 2024 macros version or we back to the one without macros?

**Comment by feliciss** [comment]
> > > Rooch move do not support macros yet.
> > 
> > 
> > Is there a way to support the 2024 macros version or we back to the one without macros?
> 
> Back to the one without macros.

I have submitted the ones without macros.

**Review by jolestar — COMMENTED** [review]
Needs some testcase.

**Comment by feliciss** [comment]
> Needs some testcase.

Added.

**Review by jolestar — APPROVED** [review]

**Pull request merged**